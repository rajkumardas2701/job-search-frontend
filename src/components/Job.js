import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Job = ({ job, user }) => (
  <>
    {console.log(user)}
    <ul>
      <li>{job.location}</li>
      <li>{job.salary}</li>
      <li>{job.role}</li>
      <li>{job.skills}</li>
      {
      (user.user.user_type === 'Candidate')
        ? (<Link to="/">View and Apply</Link>)
        : (<Link to="/">Check Applicants</Link>)
      }
    </ul>
  </>
);

Job.propTypes = {
  job: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
  user: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
};

export default Job;
