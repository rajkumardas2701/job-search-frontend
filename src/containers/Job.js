import axios from 'axios';

const Job = () => {
  const jobs = axios.get('http://localhost:3001/api/v1/jobs')
    .then((response) => {
      if (!response.data.status === 200) {
        console.log(response.data.message);
      }
      console.log(response.data.jobs);
      return response.data.jobs;
    })
    .catch((error) => console.log(error));
  return (
    <ul>
      {
        (jobs.length) ? (jobs.map((job) => <li key={job.id}>{job.location}</li>)) : ''
      }
    </ul>
  );
};

export default Job;
