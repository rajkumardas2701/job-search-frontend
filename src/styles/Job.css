.job-section {
  width: 70%;
  margin: 50px auto;
  padding: 20px;
  border: 2px solid white;
  border-radius: 10px;
  height: 300px;
  background: white;
}

.job-details {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.job-label {
  font-size: large;
  font-weight: 700;
}

.job-value {
  color: blue;
  font-weight: 500;
}

.job-btn {
  border: 1px solid black;
  margin-top: 5px;
  height: 30px;
  background: rgb(9, 219, 9);
  border-radius: 5px;
  padding-top: 4px;
}

.job-btn > a {
  text-decoration: none;
}
