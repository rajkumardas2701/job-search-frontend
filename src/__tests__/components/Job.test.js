import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { cleanup } from '@testing-library/react';
import Job from '../../components/Job';

afterEach(cleanup);

const job = {
  id: 1,
  location: 'delhi',
  salary: 12,
  role: 'tester',
  skills: 'React',
};

const user = {
  user: {
    id: 1,
    email: 'r@test.com',
    password_digest: '$2a$12$OA54/kxuiomEGCbDTSFxWe1TSKJ/if64gitiL8/P.cEUXOTd/gSyi',
    firstname: 'Raj',
    lastname: 'Test',
    created_at: '2021-06-16T03:52:25.339Z',
    updated_at: '2021-06-16T03:52:25.339Z',
  },
};

const loginState = true;

const jobMount = (
  <MemoryRouter>
    <Job
      job={job}
      user={user}
      isLoggedIn={loginState}
    />
  </MemoryRouter>
);

it('renders item component correctlly', () => {
  const job1 = renderer.create(jobMount).toJSON();
  expect(job1).toMatchSnapshot();
});
