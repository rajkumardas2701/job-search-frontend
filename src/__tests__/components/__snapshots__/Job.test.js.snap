// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders item component correctlly 1`] = `
<div
  className="job-section"
>
  <div
    className="job-details"
  >
    <p
      className="job-label"
    >
      Role:
    </p>
    <p
      className="job-value"
    >
      tester
    </p>
  </div>
  <div
    className="job-details"
  >
    <p
      className="job-label"
    >
      Location:
    </p>
    <p
      className="job-value"
    >
      delhi
    </p>
  </div>
  <div
    className="job-details"
  >
    <p
      className="job-label"
    >
      Max. salary:
    </p>
    <p
      className="job-value"
    >
      â‚¹
       
      12
       
      Lacs/A
    </p>
  </div>
  <div
    className="job-details"
  >
    <p
      className="job-label"
    >
      Skills:
    </p>
    <p
      className="job-value"
    >
      React
    </p>
  </div>
  <div
    className="job-btn"
    data-testid="text-link-test"
  >
    <a
      className="text-link"
      data-testid="text-link-test"
      href="/job_apps/1"
      onClick={[Function]}
    >
      Check Applicants
    </a>
  </div>
</div>
`;
